### DISCUSSION TITLE
Sort not working with getList

### DISCUSSION THREAD

**Discussion by Doctorthe113** [pocketbase/pocketbase#5766] [Q&A] [answered]
I don't know why but sort option isn't working. Any help is appreciated!
```js
const records = await pb.collection("message").getList(1, 100, {
    filter: `guild='${guildId}' && channel='${channelId}'`,
    sort: "created",
});
```
The message collection
![image](https://github.com/user-attachments/assets/81c5bf78-100f-4f0e-b767-e7cb2e274825)

**Comment by ganigeorgiev** [comment]
I've tested it with PocketBase v0.22.22 and v0.23.0-rc and it works correctly for me (with and without filters). We also have several automated tests, both for the query params parsing and the generated query, and they return the expected output.

From your screenshot you can also see that the table is sorted by the `created` field and the order seems correct (the Admin UI also uses the JS SDK and the same `getList` function to retrieve the table records). You can try to click on the table header and confirm that the records list is reversed.

I don't think your issue is PocketBase related and it is most likely something with your client-side code or the environment setup. If you are using Next.js make sure that what you are experiencing is not caused by its default cache.

If you still think that it is a PocketBase issue, please provide more details with:
- what do you mean by _"isn't working"_
- more complete code sample showing how you are initializing the JS SDK
- the versions of the used PocketBase executable and JS SDK
- export of your collections configuration as JSON (_Admin UI > Settings > Export collections_)
- minimal info about your setup (where is deployed/running, do you have a reverse proxy, etc.)?

**Comment by Doctorthe113** [comment]
Thanks for quick response. I am using react. To be honest the same client-side code worked before but it isn't working anymore meaning the sort parameter isn't sorting based on created. In fact, it's not doing anything `-created` isn't doing anything either.

here's how client side looks roughly
```js
export default function Chatroom() {
    const [currentChannelMsgs, setCurrentChannelMsgs] = useState([]);
    const { guildId, channelId } = useParams();
    const msgsRef = useRef(currentChannelMsgs);

    const pb = new PocketBase("http://167.179.97.209:8090");
    const author = pb.authStore.model;
    pb.autoCancellation(false);

    // Grabs old messages at the begining. Called by the useEffect at the time of mount.
    const grab_old_msgs = async () => {
        const records = await pb.collection("message").getList(1, 100, {
            filter: `guild='${guildId}' && channel='${channelId}'`,
            sort: "created",
        });
        setCurrentChannelMsgs(records.items);
    };
```
One more thing, I am now using react router whereas in the past, it was a single page app. 

I am hosting the pocketbase server on a vps without any proxy. And for client I am using vite which is also hosted on the same vps.
[pb_schema.json](https://github.com/user-attachments/files/17593585/pb_schema.json)

**Reply by ganigeorgiev** [reply]
Does the sorting work when clicking on the table header in the Admin UI?

Could you provide more details about the used executable and SDK version?

**Reply by Doctorthe113** [reply]
> Does the sorting work when clicking on the table header in the Admin UI?
> 
> Could you provide more details about the used executable and SDK version?

Click on the created header doesn't work either. It doesn't affect the sorting.

executable version: PocketBase v0.22.20
sdk version: pocketbase@0.21.5

**Reply by ganigeorgiev** [reply]
Do you have any custom hooks, eg. `onRecordsListRequest`?

**Reply by Doctorthe113** [reply]
> Do you have any custom hooks, eg. `onRecordsListRequest`?

Nope.

**Reply by Doctorthe113** [reply]
> > Does the sorting work when clicking on the table header in the Admin UI?
> > Could you provide more details about the used executable and SDK version?
> 
> Click on the created header doesn't work either. It doesn't affect the sorting.
> 
> executable version: PocketBase v0.22.20 sdk version: pocketbase@0.21.5
The order remains same
![image](https://github.com/user-attachments/assets/f5014952-3b5d-4e22-9d81-ebcdea39b38b)
![image](https://github.com/user-attachments/assets/ce18c039-dd14-4ac8-8a5d-ca8a7a0aa8cd)

**Reply by ganigeorgiev** [reply]
I've tested it again with PocketBase v0.22.20 locally and it works correctly for me (_your schema import was incomplete and it is missing the relation collections but even with mocking the result is the same_).

Either you are not reporting it properly or there is definitely something going on with your environment. I have no idea what exactly is causing it but I don't think I can help you further.

My recommendation is to try to run the executable with `--dev` and check the generated SQL queries (there should be at least one with `ORDER BY` statement).

**Reply by Doctorthe113** [reply]
> I've tested it again with PocketBase v0.22.20 locally and it works correctly for me (_your schema import was incomplete and it is missing the relation collections but even with mocking the result is the same_).
> 
> Either you are not reproting it properly or there is definitely something going on with your environment. I have no idea what exactly is causing it but I don't think I can help you further.
> 
> My recommendation is to try to run the executable with `--dev` and check the generated SQL queries (there should be at least one with `ORDER BY` statement).

Alright I will look into that, thanks for your time. Appreciate it.

**Reply by ganigeorgiev** [reply]
> I am hosting the pocketbase server on a vps without any proxy. And for client I am using vite which is also hosted on the same vps.

I don't understand what do you mean by "vite which is also hosted on the same vps" but if you have a reverse proxy such as nginx, apache, caddy, cloudflare, etc. not properly forwarding the query parameters is most likely the reason why it doesn't work even in the Admin UI.

If you still think that it is PocketBase related, test it locally, create a **minimal** reproducible repo and I'll try to investigate it but as of now I don't think me or anyone else can assist you further.

**Reply by ganigeorgiev** [reply]
I've also just noticed that the UI in your screenshots doesn't seem to be properly rendered in your case, it is missing the "info" tooltip icon in front of the relation id. It should look like this:

![relation_icon](https://i.imgur.com/1WBMBFe.png)

In any case, I'll stop engaging to avoid further confusion as there are too many unknown details to try to guess what could have gone wrong. Feel free to share a minimal repo if you are still not able to identify the culprit and I'll test it.

**Comment by Doctorthe113** [comment]
By "vite which is also hosted on the same vps", I meant I am hosting my development environment(using vite) on the same vps instead of my local machine on a different port. 
I looked thoroughly and I couldn't find any reverse proxy running or installed on my vps. 
Here's the generated sql queries when I click on the "created" header
```sql
[0.00ms] SELECT {{_admins}}.* FROM `_admins` WHERE `id`='1hrhykev4dmatjl' LIMIT 1
[0.00ms] SELECT `message`.* FROM `message` LIMIT 30
[0.00ms] SELECT COUNT(DISTINCT [[message.id]]) FROM `message`
INFO GET /api/collections/heqj4l8j9w833w5/records
```

I will test it locally in the morning and report back. If the issue persists, I will make a repo.

**Reply by ganigeorgiev** [reply, marked as answer]
There is most likely something in your environment that is stripping the query parameters. The info log should look something like:

```
INFO GET /api/collections/heqj4l8j9w833w5/records?page=1&perPage=40&sort=+created
```

You've mentioned that you don't have a reverse proxy so I don't know what could be the reason for the missing query parameters. Possible culprits:
- misbehaving browser extension, including VPN or Internet proxies apps (this could also explain the different UI rendering)
- redirects

You can try to open the browser's Inspect tool and look for the submitted request in the Network tab to investigate it. If you are sending the request with query parameters but the server somehow doesn't receive them then there is definitely something intercepting the request.

**Reply by Doctorthe113** [reply]
> There is most likely something in your environment that is stripping the query parameters. The info log should look something like:
> 
> ```
> INFO GET /api/collections/heqj4l8j9w833w5/records?page=1&perPage=40&sort=+created
> ```
> 
> You've mentioned that you don't have a reverse proxy so I don't know what could be the reason for the missing query parameters. Possible culprits:
> 
>     * misbehaving browser extension, including VPN or Internet proxies apps (this could also explain the different UI rendering)
> 
>     * redirects
> 
> 
> You can try to open the browser's Inspect tool and look for the submitted request in the Network tab to investigate it. If you are sending the request with query parameters but the server somehow doesn't receive them then there is definitely something intercepting the request.

It was vpn browser extension I had. Thanks a lot for your time. This fixed it. Do you know how I can make it work with VPN in case user has vpn on their browser. Thanks again! 😄

**Reply by ganigeorgiev** [reply]
I have no idea, sorry.

Note that every web resource that relies on query parameters will be affected by this, not just PocketBase, and unless this is a configuration settings somewhere in the specific extension, I would consider this a bug and I'd suggested to report it to the extension developer.

**Reply by Doctorthe113** [reply]
> I have no idea, sorry.
> 
> Note that every web resource that relies on query parameters will be affected by this, not just PocketBase, and unless this is a configuration settings somewhere in the specific extension, I would consider this a bug and I'd suggested to report it to the extension developer.

Alright thanks again. I would have never figured it out myself that vpn would affect sorting on my app.
